feature-libs/quote/components/details/edit/quote-details-edit.component.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { UntypedFormControl, UntypedFormGroup } from '@angular/forms';
+import { ICON_TYPE } from '@spartacus/storefront';
+
+export interface EditEvent {
+  editMode: boolean;
+  name?: string;
+  description?: string;
+  expirationTime?: Date;
+}
+
+export interface EditCard {
+  title: string;
+  paragraphs: Array<{
+    title: string;
+    text: string;
+    isTextArea?: boolean;
+    charactersLimit?: number;
+  }>;
+}
+
+@Component({
+  selector: 'cx-quote-details-edit',
+  templateUrl: './quote-details-edit.component.html',
+})
+export class QuoteDetailsEditComponent implements OnInit {
+  iconTypes = ICON_TYPE;
+  editForm: UntypedFormGroup;
+
+  @Output()
+  editCard: EventEmitter<EditEvent> = new EventEmitter();
+  @Output()
+  cancelCard: EventEmitter<boolean> = new EventEmitter();
+
+  @Input()
+  content: EditCard | null;
+
+  /**
+   * Cancels the view of the edit card tile
+   * by throwing the edit event with the edit mode set to 'false'.
+   */
+  cancel(): void {
+    this.cancelCard.emit(false);
+  }
+
+  /**
+   * Edits the card tile by throwing the edit event
+   * with the edit mode set to 'false' and the edited data.
+   */
+  edit(): void {
+    const event: EditEvent = {
+      editMode: false,
+    };
+
+    Object.keys(this.editForm.controls).forEach((control) => {
+      if (this.editForm.get(control)?.touched) {
+        const value = this.editForm.get(control)?.value;
+        if (value) {
+          Object.defineProperty(event, control, {
+            value: value,
+          });
+        }
+        this.editForm.get(control)?.markAsUntouched();
+        this.editForm.get(control)?.markAsPristine();
```

Comment:
for what do we need this?

Reply:
We want to emit only the updated data and not the whole form.
After the updated data has been emitted, the corresponding fields should be set to untouched.